### PR TITLE
feat(git-prompt): add option to show upstream branch

### PR DESCRIPTION
**Pull request by lochel**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

The upstream branch will be displayed if `ZSH_THEME_GIT_SHOW_UPSTREAM` is defined.

The prompt may look like the following:

- `(prompt -> origin/prompt ↑3|✚1)`: on branch `prompt`, ahead of remote branch `origin/prompt` by 3 commits, 1 file changed but not staged
- `(prompt|●2)`: branch `prompt` isn't set up with an upstream branch

### CODE REVIEW

**Review by carlosala — COMMENTED** [review]
In my opinion this is too much verbose. Almost always a branch will point to origin/branch. Why do you think this is important/valuable?

**Comment by lochel** [comment]
In my experience the origin remote is usually read-only and mostly used to branch off feature branches. Such features branches are then usually connected to some kind of fork with write access. Also reviewing pull requests is easier when the upstream branch is visible in the terminal. 

And this new feature is optional and disabled by default, so it shouldn't bother people not interested in it.

**Comment by carlosala** [comment]
Nice, then let's go for it! Could you update the readme?

File: plugins/git-prompt/git-prompt.plugin.zsh [hunk 1]
```diff
@@ -101,6 +107,8 @@ ZSH_THEME_GIT_PROMPT_AHEAD="%{↑%G%}"
 ZSH_THEME_GIT_PROMPT_UNTRACKED="%{$fg[cyan]%}%{…%G%}"
 ZSH_THEME_GIT_PROMPT_STASHED="%{$fg_bold[blue]%}%{⚑%G%}"
 ZSH_THEME_GIT_PROMPT_CLEAN="%{$fg_bold[green]%}%{✔%G%}"
+#ZSH_THEME_GIT_SHOW_UPSTREAM=1
```
Comment (carlosala) [inline plugins/git-prompt/git-prompt.plugin.zsh:110]: Could you delete this line? Thanks!

**Review by carlosala — CHANGES_REQUESTED** [review]

**Comment by carlosala** [comment]
Thanks for your contribution!

**Pull request merged**